pysatSpaceWeather/instruments/sw_kp.py

```diff
@@ -352,63 +459,109 @@ def download(date_array, tag, inst_id, data_path):
+                               " data type being deprecated. Switch to using",
+                               " 'def' for definitive Kp or 'now' for Kp ",
+                               "nowcasts from GFZ. Downloads are no longer ",
+                               "supported by GFZ."]),
+                      DeprecationWarning, stacklevel=2)
+    elif tag in ['def', 'now']:
+        # Set the page for the definitive or nowcast Kp
+        burl = ''.join(['https://datapub.gfz-potsdam.de/download/10.5880.Kp.',
+                        '0001/Kp_', 'nowcast' if tag == 'now' else 'definitive',
+                        '/'])
+        data_cols = ['Bartels_solar_rotation_num',
+                     'day_within_Bartels_rotation', 'Kp', 'daily_Kp_sum', 'ap',
+                     'daily_Ap', 'Cp', 'C9']
+        hours = np.arange(0, 24, 3)
+        kp_translate = {'0': 0.0, '3': 1.0 / 3.0, '7': 2.0 / 3.0}
         dnames = list()
 
         for dl_date in date_array:
-            fname = 'kp{year:02d}{month:02d}.tab'
-            fname = fname.format(year=(dl_date.year
-                                       - (dl_date.year // 100) * 100),
-                                 month=dl_date.month)
-            local_fname = fname
-            saved_fname = os.path.join(data_path, local_fname)
+            fname = 'Kp_{:s}{:04d}.wdc'.format(tag, dl_date.year)
             if fname not in dnames:
-                try:
-                    logger.info(' '.join(('Downloading file for',
-                                          dl_date.strftime('%b %Y'))))
-                    sys.stdout.flush()
-                    ftp.retrbinary('RETR ' + fname,
-                                   open(saved_fname, 'wb').write)
-                    dnames.append(fname)
-                except ftplib.error_perm as exception:
-
-                    if str(exception.args[0]).split(" ", 1)[0] != '550':
-                        # Leaving a bare raise below so that ftp errors
-                        # are properly reported as coming from ftp
-                        # and gives the correct line number.
-                        # We aren't expecting any 'normal' ftp errors
-                        # here, other than a 550 'no file' error, thus
-                        # accurately raising FTP issues is the way to go
-                        raise
-                    else:
-                        # File isn't actually there, just let people know
-                        # then continue on
-                        os.remove(saved_fname)
-                        logger.info(' '.join(('File not available for',
-                                              dl_date.strftime('%x'))))
-
-        ftp.close()
+                logger.info(' '.join(('Downloading file for',
+                                      dl_date.strftime('%Y'))))
+                furl = ''.join([burl, fname])
+                req = requests.get(furl)
```

Comment:
I'm guessing that requests handles things more gracefully on its own, thus the try is no longer needed?

No feedback if the file is not available for download?

Reply:
The try was from the ftp, the http stuff is much simpler code.  As seen elsewhere, you'll get a JSON error if trying to download a file that doesn't exist.  But that is treated as an acceptable error in most places.

Reply:
So a JSON error pops up but the code continues to run? If so, then I'm fine.